### PR TITLE
Stop calling `setup.py`, use a build frontend

### LINKED ISSUE #1320: pip install works, setuptools doesn't (Windows Python 3.6.8)
I'm finding that installing pywin32 using pip succeeds whereas listing it as a dependency using the  `install_requires` argument to `setup` in a setup.py file fails.  This is on Windows 10 using Python 3.6.8 64 bit and pywin32==224.

My steps:

**Pip install**
1. Set up an empty virtual env
2.  `pip install pywin32`
3. `python -c "import win32api"`

Results in no output - i.e. successful imoprt

**Setuptools**

1. Set up another fresh virtualenv
2.  create a minimal setup.py consisting of:
```
from setuptools import setup

setup(name='testpackage',
      version=1,
      install_requires=[
          "pywin32",
      ])
```
3. at shell run `python setup.py install`
4. `python -c "import win32api"`
5.  On my machine this results in 
```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'win32api'
```

### LINKED ISSUE #1487: setup.py build fails on Visual Studio 2019 with WDK 10
Hi,
I'm facing issues with building the package using VS 2019
Could you please add support to WDK 10 or or send instructions on how to do it?

### LINKED ISSUE #2208: Running `setup.py` as a script and its use as a CLI tool are deprecated
https://packaging.python.org/en/latest/discussions/setup-py-deprecated/#is-setup-py-deprecated
> `python setup.py` and the use of setup.py as a command line tool are deprecated

https://setuptools.pypa.io/en/latest/deprecated/commands.html#running-setuptools-commands
> running python setup.py directly as a script is considered deprecated.

pywin32 itself should move away from it, I have yet to investigate everything needed for pywin32 to move away from it.

Relates to #2119

To stop using `python setup.py`, the `--skip-verstamp` flag needs to be changed. Either trying to pass it through:
- [--config-settings](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-C) and [--build-option / --global-option ](https://setuptools.pypa.io/en/latest/history.html#id281)
- Environment variables
- Change version stamping to not have a bootstrapping issue in the first place, rendering this flag obsolete.
- or another mechanism I'm not aware of

Being able to build / install from `pip install` will open the door to:
- Building / installing the latest in-development version directly from a GitHub branch https://github.com/mhammond/pywin32/issues/2306
- editable installs, which itself:
  - Allows developer to test/import modules directly from the repo https://setuptools.pypa.io/en/latest/userguide/development_mode.html
  - Helps static checkers be more aware of which files should be present, during development. https://github.com/mhammond/pywin32/blob/9ef506138adc28b848cc537a9dc64cd18d2fcabd/pyrightconfig.json#L45-L51

### PR DESCRIPTION
**Pull request by Avasam**
Closes #2208
Closes #1320 & closes #1487 by virtue of no longer supporting the install method the users were having issues with.

~~This is dependant on https://github.com/mhammond/pywin32/pull/2390~~ Done

Please test carefully. Builds and wheels should be the same.

### CODE REVIEW

File: Pythonwin/pywin/framework/mdi_pychecker.py [hunk 1]
```diff
@@ -362,8 +362,8 @@ def threadPycheckerRun(self):
                 result = "Can't find python.exe!\n"
             elif not os.path.isfile(pychecker):
                 result = (
-                    "Can't find checker.py - please install pychecker "
-                    "(or run 'setup.py install' if you have the source version)\n"
+                    "Can't find checker.py - please install PyChecker "
+                    "(https://pypi.org/project/PyChecker/)\n"
```
Comment (Avasam) [inline Pythonwin/pywin/framework/mdi_pychecker.py:366]: This code is getting removed by https://github.com/mhammond/pywin32/pull/2412 anyway

**Comment by Avasam** [comment]
Looks like all the artifacts are there
image

I wonder if the repeated `--config-setting=--build-option=--plat-name=win-arm64` flag is really necessary

**Review by mhammond — APPROVED** [review]
This seems great, thanks!

**Comment by Avasam** [comment]
I'm feeling confident about this. I'll investigate whether all the flags are really necessary separately. This is a milestone in removing legacy/deprecated code from pywin32. It also opens the door for editable installs, different build frontends, etc.

**Pull request merged**